### ISSUE TITLE
Foundations: Installations: Update external dual boot instructions

### ISSUE THREAD

**Issue by scheals** [TheOdinProject/curriculum#25026]
## Because
Current dual boot instructions we link to use Balena Etcher and go with manually creating the partitions - root partition having 16GB which won't do well in the longer run. I would like to suggest [instructions from It's FOSS](https://itsfoss.com/install-ubuntu-1404-dual-boot-mode-windows-8-81-uefi/) they use Rufus, suggest at least 25GB for root if `Install alongside Windows Boot Manager` option is not present (30-40GB+ otherwise, for some reason), they go through checking whether it is a MBR or GPT drive & whether it is a Legacy or a UEFI install. It also links to instructions for SSD & HDD install which are neat and to MBR install, which are okay-ish, more on that later.

We can also rely pretty much completely on the guide given our instructions - it'd make more sense to put them in the very beginning rather than at the end of the instructions. While we encourage (I believe) reading the instructions in full before doing them step-by-step, people like me forget to do so and I do not think I am in the minority.

Now potential drawbacks:
- The video that goes over the install uses Balena Etcher for some reason. Possible source of confusion but the PR could make note of it. 
- Folks that don't see the `Install alongside Windows Boot Manager` option will be walked through `Something else` option and while it might be all good, in some cases it is a symptom of some mismatch, like trying to install Ubuntu on a MBR disk that doesn't have any more primary partition slots. Hopefully folks don't nuke their install and come to us instead but we might want to encourage asking us.
- [The MBR install instructions](https://itsfoss.com/install-ubuntu-dual-boot-mode-windows/) might pose somewhat of a risk in step 4:
  > As you can see, I have 3 NTFS and some ext4 partitions. If you don’t have ext4 partition, don’t worry, we don’t need that. As you can see in the picture below, one of the NTFS partition consists of Windows installation. This should be untouched if you want to keep your Windows installation safe. 
  I presume that you have more than one NTFS (or FAT 32) partitions (i.e. Drives in Windows term) on your hard disk, with one of them consisting of Windows installation (usually C drive). What you need to do here is to delete a NTFS or existing ext4 partition and create some free space. This will delete all the data in that partition and this is why I asked you to verify if you have Windows installed in a different partition.

  I can easily see people nuking their recovery/boot partitions because at least in the screenshot provided, they show up as NTFS. So two ways here: add a note about EXTREME caution around that bit OR we in-house some basic instructions for MBR2GPT since the official ones are pretty darn good and then we don't have to worry about those MBR install instructions. The only complexity with MBR2GPT instructions would be figuring out whether someone has a primary partition slot available, because it will fail if they don't. Could throw that into the PR as well.
  
  ## Proposed implementation
  - Swap the links.
  - *Optional* Move the link to the beginning and make it Step 2 of current instructions while foregoing the following ones.
  - *Optional* Make note of the Balena Etcher mismatch in the YouTube video provided by the instructions.
  - *Optional* Make a note against potentially nuking your recovery/boot partitions.
  - *Optional* In-house some MBR2GPT instructions so folks don't have to rely on our notes (that we know people forget all the time) & they can use the original, UEFI GPT set of instructions.
  - *Optional* Keep folks that have a Legacy-only BIOS in mind and mention the MBR installation instructions anyway with the cautionary note.

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-installations#os-installation

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

**Comment by wise-king-sullyman** [comment]
Do you have a particular idea of what you'd want to put together? @xandora suggested in our issue combing session that it would probably be next to impossible to create a perfect solution for all possible setups.

**Comment by scheals** [comment]
@wise-king-sullyman It's been a while since I have helped with these issues but quick skim of this issue gets me to:
- Find a better guide than RoboShack one, link it instead - with caveats or without,
- include some MBR2GPT guide, potentially an in-house one, could try looking through past convos about the process on Discord to build it.

**Comment by wise-king-sullyman** [comment]
@xandora how does that plan sound to you?

**Comment by ManonLef** [comment]
@xandora you might have missed the ping up there :)

**Comment by xandora** [comment]
@ManonLef Ha... so this is why I have a half finished dual boot guide in my Obsidian vault...

**Comment by ManonLef** [comment]
@xandora we could keep it as a discussion for a while. Do you reckon the amount of dual boots to go down with the addition of WSL2?

**Comment by scheals** [comment]
@xandora I'm ready to come back to this. I need to keep #25812 in mind as well.

**Comment by thatblindgeye** [comment]
@xandora is this still something we'd want to implement?

**Comment by AirDevil188** [comment]
I would like to work on this issue if the original author doesn't  want to work on it anymore.

**Comment by JoshDevHub** [comment]
I definitely have some thoughts about this.

A full in-house guide seems like a lot of upfront work and maintenance for an option I'd wager few of our learners use. I think we should either have some small corrections for the guide we link (for example, telling users not to use 16GB for root) or just axe the dual boot instructions altogether.

I think I might actually be partial to the second option now that we support WSL, but also interested in what others think.

**Comment by scheals** [comment]
I'm happy to let @AirDevil188 work on this - I might not get to this as fast as Uki would 😄. I still think the It's FOSS guide is great just have to keep in mind the potential pitfalls that I've noted.

Folks that have old systems are more likely to opt for a dual boot and more likely to encounter issues with MBR, while [Microsoft's docs are pretty great](https://learn.microsoft.com/en-us/windows/deployment/mbr-to-gpt) I'm a huge proponent of reducing the friction in the installation section as much as possible - meaning, hand-hold through the process and warn about potential pitfalls. This is obviously even more important with dual booting because we're risking losing some of data or all of it + the OS.

But yeah, just not having to tell people that the guide TOP links to is not that great because of space suggested every time dual booting comes up is going to be a huge time saver for both parties. And obviously, folks that are not on Discord are going to avoid this pitfall due to changing the guide/commenting on that space allocation for root.

**Comment by AirDevil188** [comment]
I think even though that we now support WSL2 I don't think that we should `axe` the dual boot instructions altogether. Because WSL2 is in my eyes option that is more for the users that are more advanced. If I remember right even our curriculum makes note on that. A lot of new users will make a common mistake thinking that WSL2 is `all Windows`. When in reality it's not Windows at all. They are just looking   at  `Linux/Ubuntu` VM through  Windows lenses. 
 
 The more I think about this the more I'm leaning towards the `adding a note/warning  solution` before 
 Dave's RoboShacks guide - to leave 30gb for `root `  `/` instead of 16gb.

**Comment by xandora** [comment]
I'm not terribly in favour of peppering the curriculum with little notes to cover all the gotchas people can run into, but I think leaving a note is probably a good temporary solution to the problem.

**Comment by AirDevil188** [comment]
I have read through the  FOSS guide that @scheals has linked, and I also think that it's pretty great, and it's not hard to follow. The only thing that I can see that would confuse users is the part about checking whatever tue partition is mbr or gpt. I'm not sure if this would be a potential pitfall for a beginner.

**Comment by wise-king-sullyman** [comment]
@TheOdinProject/installations can you all investigate this? Is it something we still need to keep open?

**Comment by zachmmeyer** [comment]
> [@TheOdinProject/installations](https://github.com/orgs/TheOdinProject/teams/installations) can you all investigate this? Is it something we still need to keep open?

🔎 Investigating.

**Comment by mao-sz** [comment]
@zachmmeyer with #31333 complete, is this still relevant?

**Comment by zachmmeyer** [comment]
Nope, no longer relevant! Using the official guides covers all of this.
